app/kubernetes-ingress-controller/reference/feature-gates.md

```diff
@@ -0,0 +1,165 @@
+The same definitions of `feature gates` and `feature stages` from upstream Kubernetes are used to define KIC's list of features.
+
+* [Kubernetes](https://kubernetes.io)
+* [Kubernetes feature gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/)
+* [Feature stages](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-stages)
+
+## Available feature gates
+
+{% table %}
+columns:
+  - title: Feature
+    key: feature
+  - title: Default
+    key: default
+  - title: Stage
+    key: stage
+  - title: Since
+    key: since
+  - title: Until
+    key: until
+rows:
+  - feature: GatewayAlpha
+    default: '`false`'
+    stage: Alpha
+    since: 2.6.0
+    until: TBD
+  - feature: FillIDs
+    default: '`true`'
+    stage: Beta
+    since: 3.0.0
+    until: TBD
+  - feature: RewriteURIs
+    default: '`false`'
+    stage: Alpha
+    since: 2.12.0
+    until: TBD
+  - feature: KongServiceFacade
+    default: '`false`'
+    stage: Alpha
+    since: 3.1.0
+    until: TBD
+  - feature: SanitizeKonnectConfigDumps
+    default: '`true`'
+    stage: Beta
+    since: 3.1.0
+    until: TBD
+  - feature: FallbackConfiguration
+    default: '`false`'
+    stage: Alpha
+    since: 3.2.0
+    until: TBD
+  - feature: KongCustomEntity
+    default: '`false`'
+    stage: Alpha
+    since: 3.2.0
+    until: 3.3.0
+  - feature: KongCustomEntity
+    default: '`true`'
+    stage: Beta
+    since: 3.3.0
+    until: 3.4.0
+  - feature: KongCustomEntity
+    default: '`true`'
+    stage: GA
+    since: 3.4.0
+    until: TBD
+  - feature: CombinedServicesFromDifferentHTTPRoutes
+    default: '`false`'
+    stage: Alpha
+    since: 3.4.0
+    until: TBD
+{% endtable %}
+
+
+## Using feature gates
+
+To enable feature gates, provide the `--feature-gates` flag when launching KIC, or set the `CONTROLLER_FEATURE_GATES` environment variable.
+
+Feature gates consist of a comma-delimited set of `key=value` pairs. For example, if you wanted to enable `FillIDs` and `RewriteURIs`, you'd set `CONTROLLER_FEATURE_GATES=FillIDs=true,RewriteURIs=true`.
+
+To enable features via Helm, set the following in your `values.yaml`:
+
+{% navtabs chart %}
+{% navtab "kong/ingress" %}
+```yaml
+controller:
+  ingressController:
+    env:
+      feature_gates: FillIDs=true,RewriteURIs=true
+```
+{% endnavtab %}
+{% navtab "kong/kong" %}
+```yaml
+ingressController:
+  env:
+    feature_gates: FillIDs=true,RewriteURIs=true
+```
+{% endnavtab %}
+{% endnavtabs %}
+
+To test a feature gate in an existing deployment, use `kubectl set env`.
+
+```bash
+kubectl set env -n kong deployment/kong-controller \
+  CONTROLLER_FEATURE_GATES="FillIDs=true,RewriteURIs=true" \
+  -c ingress-controller
+```
+
+## Feature gate availability
+
+* The **since** and **until** rows in the table above refer to [KIC Releases](https://github.com/Kong/kubernetes-ingress-controller/releases).
```

Comment:
```suggestion
* The **since** and **until** rows in the [feature gates table](#available-feature-gates) refer to [KIC Releases](https://github.com/Kong/kubernetes-ingress-controller/releases).
```